database/factories/BeatmapsetFactory.php

```diff
@@ -31,7 +32,7 @@ public function definition(): array
             'genre_id' => Genre::factory(),
             'language_id' => Language::factory(),
             'submit_date' => fn () => $this->faker->dateTime(),
-            'thread_id' => 0,
+            'thread_id' => Topic::factory()->withPost(),
```

Comment:
Beatmapset with description should be a separate state, we don't want to create an extra post and related stuff with the default factory when they're not needed.